lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -690,35 +713,53 @@ static Operation *generateFMAOp(vector::FMAOp fmaOp, AIEOpAttributes &opAttr,
                    .getElementType()
                    .isa<IntegerType>();
 
-  if (isInt && !writesToAccumulator(acc.getDefiningOp())) {
+  if ((isInt || AIEML) && !writesToAccumulator(acc.getDefiningOp())) {
     acc = generateUPSOp(acc, state, fmaOp->getLoc());
     LLVM_DEBUG(llvm::dbgs()
                << "\n\nCreated UPS op " << acc << " to move the output of "
                << fmaOp << " into accumulator");
   }
   // If the lhs operand vector is not >= twice the rhs operand vector, then use
-  // concat operator.
+  // concat operator. If targeting for AIE-ML intrinsics, use broadcast operator
+  // for rhs.
   Value lhs = fmaOp.getLhs();
+  Value rhs = fmaOp.getRhs();
```

Comment:
Hi @linay-xsj : 
I see little commonality between AIEML and !AIEML from 727 till the end of the function.
I suggest that 727-764 be recoded as:
`Operation *xfmaOp = nullptr;
 if (AIEML) {
   .. 
  } else { 
   .. 
  }
`
This will make it clear which cases are currently supported for AIE and which cases are currently supported for AIEML.